File: qqbot-protocol/src/main/java/com/mieai/qqbot/protocol/openapi/QqGuildModels.java
package com.mieai.qqbot.protocol.openapi;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/** Wire models shared by QQ guild, channel, member, and role OpenAPI operations. */
public final class QqGuildModels {
    private QqGuildModels() {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record User(
            String id,
            String username,
            String avatar,
            Boolean bot,
            @JsonProperty("union_openid") String unionOpenId,
            @JsonProperty("union_user_account") String unionUserAccount) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Guild(
            String id,
            String name,
            String icon,
            @JsonProperty("owner_id") String ownerId,
            Boolean owner,
            @JsonProperty("member_count") Integer memberCount,
            @JsonProperty("max_members") Long maxMembers,
            String description,
            @JsonProperty("joined_at") String joinedAt,
            List<Channel> channels,
            @JsonProperty("union_world_id") String unionWorldId,
            @JsonProperty("union_org_id") String unionOrgId,
            @JsonProperty("op_user_id") String operatorUserId) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Channel(
            String id,
            @JsonProperty("guild_id") String guildId,
            String name,
            Integer type,
            @JsonProperty("sub_type") Integer subType,
            Long position,
            @JsonProperty("parent_id") String parentId,
            @JsonProperty("owner_id") String ownerId,
            @JsonProperty("private_type") Integer privateType,
            @JsonProperty("speak_permission") Integer speakPermission,
            @JsonProperty("application_id") String applicationId,
            String permissions,
            @JsonProperty("op_user_id") String operatorUserId) {}

    public record ChannelUpdate(
            String name,
            Integer type,
            @JsonProperty("sub_type") Integer subType,
            Long position,
            @JsonProperty("parent_id") String parentId,
            @JsonProperty("private_type") Integer privateType,
            @JsonProperty("private_user_ids") List<String> privateUserIds,
            @JsonProperty("speak_permission") Integer speakPermission,
            @JsonProperty("application_id") String applicationId) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Member(
            @JsonProperty("guild_id") String guildId,
            @JsonProperty("joined_at") String joinedAt,
            String nick,
            User user,
            List<String> roles,
            @JsonProperty("op_user_id") String operatorUserId) {}

    public record MemberDeleteRequest(
            @JsonProperty("add_blacklist") boolean addBlacklist,
            @JsonProperty("delete_history_msg_days") int deleteHistoryMessageDays) {}

    public record MemberRoleRequest(Channel channel) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Role(
            String id,
            String name,
            Long color,
            Integer hoist,
            Integer number,
            @JsonProperty("member_limit") @JsonAlias("number_limit") Integer memberLimit) {}

    public record RoleUpdate(String name, Long color, Integer hoist) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record GuildRoles(
            @JsonProperty("guild_id") String guildId,
            List<Role> roles,
            @JsonProperty("role_num_limit") String roleNumberLimit) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record RoleUpdateResult(
            @JsonProperty("guild_id") String guildId,
            @JsonProperty("role_id") String roleId,
            Role role) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record RoleMembersPage(List<Member> data, String next) {}

    public record MuteRequest(
            @JsonProperty("mute_end_timestamp") String muteEndTimestamp,
            @JsonProperty("mute_seconds") String muteSeconds,
            @JsonProperty("user_ids") List<String> userIds) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record MuteResult(@JsonProperty("user_ids") List<String> userIds) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record MessageSetting(
            @JsonProperty("disable_create_dm") Boolean disableCreateDirectMessage,
            @JsonProperty("disable_push_msg") Boolean disablePushMessage,
            @JsonProperty("channel_ids") List<String> channelIds,
            @JsonProperty("channel_push_max_num") Integer channelPushMaximum) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record OnlineMemberCount(@JsonProperty("online_nums") Integer onlineMembers) {}
}

File: qqbot-protocol/src/main/kotlin/com/mieai/qqbot/protocol/openapi/QqGuildModels.kt
package com.mieai.qqbot.protocol.openapi

import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

/** Wire models shared by QQ guild, channel, member, and role OpenAPI operations. */
object QqGuildModels {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class User(
        val id: String?,
        val username: String?,
        val avatar: String?,
        val bot: Boolean?,
        @JsonProperty("union_openid") val unionOpenId: String?,
        @JsonProperty("union_user_account") val unionUserAccount: String?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Guild(
        val id: String?,
        val name: String?,
        val icon: String?,
        @JsonProperty("owner_id") val ownerId: String?,
        val owner: Boolean?,
        @JsonProperty("member_count") val memberCount: Int?,
        @JsonProperty("max_members") val maxMembers: Long?,
        val description: String?,
        @JsonProperty("joined_at") val joinedAt: String?,
        val channels: List<Channel>?,
        @JsonProperty("union_world_id") val unionWorldId: String?,
        @JsonProperty("union_org_id") val unionOrgId: String?,
        @JsonProperty("op_user_id") val operatorUserId: String?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Channel(
        val id: String?,
        @JsonProperty("guild_id") val guildId: String?,
        val name: String?,
        val type: Int?,
        @JsonProperty("sub_type") val subType: Int?,
        val position: Long?,
        @JsonProperty("parent_id") val parentId: String?,
        @JsonProperty("owner_id") val ownerId: String?,
        @JsonProperty("private_type") val privateType: Int?,
        @JsonProperty("speak_permission") val speakPermission: Int?,
        @JsonProperty("application_id") val applicationId: String?,
        val permissions: String?,
        @JsonProperty("op_user_id") val operatorUserId: String?,
    )

    data class ChannelUpdate(
        val name: String?,
        val type: Int?,
        @JsonProperty("sub_type") val subType: Int?,
        val position: Long?,
        @JsonProperty("parent_id") val parentId: String?,
        @JsonProperty("private_type") val privateType: Int?,
        @JsonProperty("private_user_ids") val privateUserIds: List<String>?,
        @JsonProperty("speak_permission") val speakPermission: Int?,
        @JsonProperty("application_id") val applicationId: String?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Member(
        @JsonProperty("guild_id") val guildId: String?,
        @JsonProperty("joined_at") val joinedAt: String?,
        val nick: String?,
        val user: User?,
        val roles: List<String>?,
        @JsonProperty("op_user_id") val operatorUserId: String?,
    )

    data class MemberDeleteRequest(
        @JsonProperty("add_blacklist") val addBlacklist: Boolean,
        @JsonProperty("delete_history_msg_days") val deleteHistoryMessageDays: Int,
    )

    data class MemberRoleRequest(val channel: Channel?)

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Role(
        val id: String?,
        val name: String?,
        val color: Long?,
        val hoist: Int?,
        val number: Int?,
        @JsonProperty("member_limit") @JsonAlias("number_limit") val memberLimit: Int?,
    )

    data class RoleUpdate(val name: String?, val color: Long?, val hoist: Int?)

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class GuildRoles(
        @JsonProperty("guild_id") val guildId: String?,
        val roles: List<Role>?,
        @JsonProperty("role_num_limit") val roleNumberLimit: String?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class RoleUpdateResult(
        @JsonProperty("guild_id") val guildId: String?,
        @JsonProperty("role_id") val roleId: String?,
        val role: Role?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class RoleMembersPage(val data: List<Member>?, val next: String?)

    data class MuteRequest(
        @JsonProperty("mute_end_timestamp") val muteEndTimestamp: String?,
        @JsonProperty("mute_seconds") val muteSeconds: String?,
        @JsonProperty("user_ids") val userIds: List<String>?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class MuteResult(@JsonProperty("user_ids") val userIds: List<String>?)

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class MessageSetting(
        @JsonProperty("disable_create_dm") val disableCreateDirectMessage: Boolean?,
        @JsonProperty("disable_push_msg") val disablePushMessage: Boolean?,
        @JsonProperty("channel_ids") val channelIds: List<String>?,
        @JsonProperty("channel_push_max_num") val channelPushMaximum: Int?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class OnlineMemberCount(@JsonProperty("online_nums") val onlineMembers: Int?)
}
